framework/src/bcs/ArrayDirichletBC.C

```diff
@@ -31,8 +31,8 @@ ArrayDirichletBC::ArrayDirichletBC(const InputParameters & parameters)
 {
 }
 
-RealEigenVector
-ArrayDirichletBC::computeQpResidual()
+void
+ArrayDirichletBC::computeQpResidual(RealEigenVector & residual)
 {
-  return _u - _values;
+  residual = _u - _values;
```

Comment:
Can you add a check in the constructor on the size of `_values` vs `_var.count()`?